### ISSUE TITLE
Walking Distance in Routing

### ISSUE THREAD

**Issue by LeoWoerteler** [JosuaKrause/BusVis#31]
The routing algorithm should also use the walking distance between stations, e.g. from the [Google Directions API](https://developers.google.com/maps/documentation/directions/).

**Comment by LeoWoerteler** [comment]
Progress can be seen at https://github.com/LeoWoerteler/BusVis/compare/gh-31 .

**Comment by LeoWoerteler** [comment]
## Before:

```
RoutingResult[
  from=Zähringerplatz,
  to=Sternenplatz / Spanierstraße,
  steps=[
    Start at Zähringerplatz,
    wait for 0 minutes,
    take bus line 3 to Schottenplatz,
    wait for 3 minutes,
    take bus line 9 to Ellenrieder Gymnasium,
    wait for 8 minutes,
    take bus line 6 to Sternenplatz / Spanierstraße
  ],
  time=20 min
]
```

## Now:

```
RoutingResult[
  from=Zähringerplatz,
  to=Sternenplatz / Spanierstraße,
  steps=[
    Start at Zähringerplatz,
    wait for 0 minutes,
    take bus line 3 to Sternenplatz,
    walk 2 min to Sternenplatz / Spanierstraße
  ],
  time=4 min
]
```

**Comment by JosuaKrause** [comment]
Is this the `toString` method of `RoutingResult`? nice!
Seems `RoutingResult#getEdges()` may contain null pointer edges. Brakes `EdgeMatrix` for me. null pointer edges should not be in the `RoutingResult`.
EDIT: I'm sorry the null pointers are the `UndirectedEdges`. Maybe Walking edges are not added in the `EdgeMatrix`?
EDIT2: Why not adding walking distances as edges in bus stations?
EDIT3: I see now... But you have to encode it in the edge matrix somehow...

**Comment by LeoWoerteler** [comment]
I think only used walking edges should be shown in the visualization, otherwise it would get far too cluttered. Do you think that all edges should be in the `EdgeMatrix`?

**Comment by JosuaKrause** [comment]
LeoWoerteler#2 fixes the issue. It gives a little overhead but makes it possible to highlight any connection between two arbitrary connections (that are not the same).

**Comment by JosuaKrause** [comment]
Walking is kind of broken (see LeoWoerteler#2 for details). The pull request is fine though since it does not alter the routing algorithm.

**Comment by LeoWoerteler** [comment]
The routing bug should be fixed, I forgot to add walking edges at the very beginning.
I also added control elements to the GUI.

**Comment by JosuaKrause** [comment]
implemented by #32
